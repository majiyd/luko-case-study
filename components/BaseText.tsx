import React from 'react'
import {Text, TextStyle, StyleSheet } from 'react-native'
import Colors from '../constants/Colors';

type Props = {
  type?: 'headline' | 'sub-body' | 'body' | 'caption';
  children: React.ReactNode;
  style?: TextStyle
};
const BaseText = ({children, type="sub-body", style}: Props) => {
  if (type === 'headline') {
    return <Text style={[styles.headline, style]}>{children}</Text>
  
  }
  return (
      <Text style={[styles.subBody, style]}>{children}</Text>
  )
}

const styles = StyleSheet.create({
  headline: {
    fontSize: 19,
    fontWeight: 'bold',
    lineHeight: 26,
    color: Colors.light.gray1000
  },
  subBody: {
    fontSize: 15,
    lineHeight: 20,
    color: Colors.light.gray700
  }
})
export default BaseText
