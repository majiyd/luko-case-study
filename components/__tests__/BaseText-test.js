import * as React from "react";
import renderer from "react-test-renderer";
import BaseText from "../BaseText";

it(" InventoryItem renders correctly", () => {
  const tree = renderer.create(<BaseText>HIIII</BaseText>).toJSON();
  expect(tree).toMatchSnapshot();
});
