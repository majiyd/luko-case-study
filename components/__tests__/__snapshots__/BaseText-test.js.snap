// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` InventoryItem renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#6B6651",
        "fontSize": 15,
        "lineHeight": 20,
      },
      undefined,
    ]
  }
>
  HIIII
</Text>
`;
